DataDog/libdatadog
# feat(profiling)!: pass errors through more FFI functions

# What does this PR do?

This changes the return types for FFI functions:
 - `ddog_prof_Profile_add`
 - `ddog_prof_Exporter_Request_build`

Adds new structs:
 - `ddog_prof_Profile_AddResult`
 - `ddog_prof_Exporter_Request_BuildResult`
 
And adds functions to drop them:
 - `ddog_prof_Profile_AddResult_drop`
 - `ddog_prof_Exporter_Request_BuildResult_drop`

Removes a now-unnecessary newtype definition of `struct Request(exporter::Request)` as cbindgen handles the refactored code.

# Motivation

As we will be releasing v2.0 soon, I figured we could ship some other breaking changes that help FFI users know _what_ went wrong, rather than just that something did.

# Additional Notes

Please audit the names as per the recent naming scheme names.

# How to test the change?

Download the artifacts, adjust the code surrounding the two functions listed above, and ensure it all still works correctly.
